glide-core/src/client/types.rs

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+ */
+
+#[derive(Default)]
+pub struct ConnectionRequest {
```

Comment:
unless you have a very good reason for it, you shouldn't use `* c_char `in your Rust structs, only in function signatures. instantiate them to Rust strings ASAP instead of passing unsafe pointers through the system.

Reply:
let's decide if we want this, get this in, and then make the adjustments to make if fully FFI-compatible. I don't want to overload this PR with multiple changes.